paasta_tools/utils.py

```diff
@@ -3013,7 +3013,15 @@ def get_pipeline_config(service: str, soa_dir: str = DEFAULT_SOA_DIR) -> List[Di
 def get_pipeline_deploy_groups(
     service: str, soa_dir: str = DEFAULT_SOA_DIR
 ) -> List[str]:
-    pipeline_steps = [step["step"] for step in get_pipeline_config(service, soa_dir)]
+    pipeline_steps = []
+    for step in get_pipeline_config(service, soa_dir):
+        if(step.get("parallel", False)):
```

Comment:
if you install/run the pre-commit hooks, it'll probably re-write this to 
```suggestion
        if step.get("parallel", False):
```

(and same below)

that said, should we rely on the default `get()` behavior of returning `None` if the key doesn't exist? doesn't seem like we explicitly need a boolean and None is falsey :)